resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerUtils.scala

```diff
@@ -80,10 +80,27 @@ trait MesosSchedulerUtils extends Logging {
     }
     fwInfoBuilder.setHostname(Option(conf.getenv("SPARK_PUBLIC_DNS")).getOrElse(
       conf.get(DRIVER_HOST_ADDRESS)))
+    conf.getOption("spark.mesos.principal.file")
+      .orElse(Option(conf.getenv("SPARK_MESOS_PRINCIPAL_FILE"))
+      .foreach { principalFile =>
+        val file = io.Source.fromFile(principalFile)
+        val principal = file.getLines.next()
+        file.close
+        fwInfoBuilder.setPrincipal(principal)
+        credBuilder.setPrincipal(principal)
+      }
     conf.getOption("spark.mesos.principal").foreach { principal =>
       fwInfoBuilder.setPrincipal(principal)
       credBuilder.setPrincipal(principal)
     }
+    conf.getOption("spark.mesos.secret.file")
+      .orElse(Option(conf.getenv("SPARK_MESOS_SECRET_FILE"))
```

Comment:
Why not provide the secret directly in the environment, instead of indirectly through a file?

That is more secure than files in certain situations, so it would be nice to at least have that option.

Reply:
Environment-variable secrets (even using the `secrets` primitive up to Mesos 1.4) will be available at `/proc/<pid>/environ` so file-based is better more secure. I'm tempted to say having an insecure _option_ is worse than having less flexibility.

Reply:
You can't read `/proc/<pid>/environ` from other users' processes.

```
$ cat /proc/1/environ
cat: /proc/1/environ: Permission denied
```